internal/cli/watch_opts.go

```diff
@@ -55,6 +60,23 @@ func (opts *WatchOpts) Watch(f Watcher) error {
 	}
 }
 
+func (opts *WatchOpts) exponentialBackoff(f Watcher) (bool, error) {
+	if opts.exponentialBackoffTrigger == nil {
+		return f()
+	}
+	backoffTimes := []time.Duration{2 * time.Second, 4 * time.Second, 8 * time.Second}
+	for _, backoffTime := range backoffTimes {
+		done, err := f()
+		if opts.exponentialBackoffTrigger(err) {
+			time.Sleep(backoffTime)
+			continue
+		}
+		return done, err
+	}
+	// Should only happen after trying three times (>14 seconds)
+	return f()
+}
```

Comment:
couple things, renaming `exponentialBackoffTrigger` to `IsRetryableErr` then similar to recrating variables in loops recreating the times is no bueno
```suggestion
var backoffTimes = []time.Duration{2 * time.Second, 4 * time.Second, 8 * time.Second}

func (opts *WatchOpts) exponentialBackoff(f Watcher) (bool, error) {
	if opts.IsRetryableErr == nil {
		return f()
	}

	for _, backoffTime := range backoffTimes {
		done, err := f()
		if err != nil && opts.IsRetryableErr(err) {
			time.Sleep(backoffTime)
			continue
		}
		return done, err
	}
	// Should only happen after trying three times (>14 seconds)
	return f()
}
```

Reply:
also optional and a matter of preference, I find the continue logic hard to read 
```suggestion
var backoffTimes = []time.Duration{2 * time.Second, 4 * time.Second, 8 * time.Second}

func (opts *WatchOpts) exponentialBackoff(f Watcher) (bool, error) {
	if opts.IsRetryableErr == nil {
		return f()
	}

	for _, backoffTime := range backoffTimes {
		if done, err := f(); err == nil || !opts.IsRetryableErr(err) {
			return done, err
		}
		time.Sleep(backoffTime)
	}
	// Should only happen after trying three times (>14 seconds)
	return f()
}
```